course3/homework-unique_ptr/my_unique_ptr.h

```diff
@@ -0,0 +1,93 @@
+#include <cstddef>
+#include <iostream>
+
+template<class T>
+class MyUniquePtr
+{
+public:
+	MyUniquePtr() : mData(nullptr)
+	{
+		std::cout << "MyUniquePtr::ctor" << std::endl;
+	}
+	
+	MyUniquePtr(MyUniquePtr& that)
+	{
+		// Construction one unique pointer from another simply changes the ownership of the data
+		mData = that.mData;
+		that.mData = nullptr;
+		std::cout << "MyUniquePtr::Copy" << std::endl;
+	}
+	
+	MyUniquePtr(T*&& data) : mData(data)
```

Comment:
and this is rvalue of a pointer to T. RValues to simple types make little sense.
Please, consult the C++ reference for the signature of unique_ptr